cmd/meroxa/turbine/golang/deploy.go

```diff
@@ -2,144 +2,119 @@ package turbinego
 	}
-
-	if err := json.Unmarshal(output, &resources); err != nil {
-		// fall back if not json
-		return utils.GetResourceNamesFromString(string(output)), nil
+	var resources []turbine.ApplicationResource
+	for _, r := range resp.Resources {
+		resources = append(resources, turbine.ApplicationResource{
+			Name:        r.Name,
+			Destination: r.Destination,
+			Source:      r.Source,
+			Collection:  r.Collection,
+		})
 	}
-
 	return resources, nil
 }
 
 // NeedsToBuild reads from the Turbine application to determine whether it needs to be built or not
 // this is currently based on the number of functions.
-func (t *turbineGoCLI) NeedsToBuild(ctx context.Context, appName string) (bool, error) {
-	cmd := exec.CommandContext(ctx, path.Join(t.appPath, appName), "--listfunctions") //nolint:gosec
-
-	accessToken, refreshToken, err := global.GetUserToken()
+// TODO: Once all languages are under turbine-core refactor this so it's the same for all languages.
+func (t *turbineGoCLI) NeedsToBuild(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+	resp, err := t.bc.HasFunctions(ctx, &emptypb.Empty{})
 	if err != nil {
 		return false, err
 	}
-	cmd.Env = os.Environ()
-	cmd.Env = append(cmd.Env, fmt.Sprintf("ACCESS_TOKEN=%s", accessToken), fmt.Sprintf("REFRESH_TOKEN=%s", refreshToken))
-
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		fmt.Println(string(output))
-		return false, fmt.Errorf("build failed")
-	}
-	list, err := utils.GetTurbineResponseFromOutput(string(output))
-	if err == nil && list != "" {
-		// ignores any lines that are not intended to be part of the response
-		list = strings.Trim(list, "[]")
-		if list == "" {
-			return false, nil
-		}
-
-		hasFunctions := len(strings.Split(list, ",")) > 0
-		return hasFunctions, nil
-	}
 
-	re := regexp.MustCompile(`\[(.+?)]`)
-	// stdout is expected as `"2022/03/14 17:33:06 available functions: []` where within [], there will be each function.
-	hasFunctions := len(re.FindAllString(string(output), -1)) > 0
-
-	return hasFunctions, nil
+	return resp.Value, nil
 }
 
 func (t *turbineGoCLI) GetGitSha(ctx context.Context) (string, error) {
-	return utils.GetGitSha(ctx, t.appPath)
+	return turbine.GetGitSha(ctx, t.appPath)
 }
 
 func (t *turbineGoCLI) GitChecks(ctx context.Context) error {
-	return utils.GitChecks(ctx, t.logger, t.appPath)
+	return turbine.GitChecks(ctx, t.logger, t.appPath)
 }
 
 func (t *turbineGoCLI) CreateDockerfile(_ context.Context, appName string) (string, error) {
 	return t.appPath, turbineGo.CreateDockerfile(appName, t.appPath)
 }
 
 func (t *turbineGoCLI) CleanUpBuild(_ context.Context) {
-	utils.CleanupDockerfile(t.logger, t.appPath)
+	turbine.CleanupDockerfile(t.logger, t.appPath)
 }
 
-func (t *turbineGoCLI) SetupForDeploy(_ context.Context, _ string) (func(), error) {
-	return func() {}, nil
+func (t *turbineGoCLI) SetupForDeploy(ctx context.Context, gitSha, appName string) (func(), error) {
```

Comment:
Still would like to refactor this method.

Reply:
I think you can totally get away with a single command like this to get the spec https://github.com/meroxa/cli/pull/711/files#diff-b5881834513c853c914f1a532fccc86b12ed20be8eaff0c074f49570c1b33754R15-R23

Reply:
compared to build binary, run binary, cleanup binary